census_view.cpp

```diff
@@ -1431,7 +2395,30 @@ void CensusView::UponRightClick(wxDataViewEvent& e)
     delete census_menu;
 }
 
-void CensusView::UponValueChanged(wxDataViewEvent&)
+void CensusGridView::UponRightClick(wxGridEvent& e)
+{
+    if(e.GetEventObject() != grid_window_)
+        {
+        // This event should come only from the window pointed to by
```

Comment:
"Should come" is not really clear, so I'd rephrase this as
```suggestion
        // We are only interested in the events originating from the main grid window
```

But, wording aside, the actual logic here is not so great because it will break if we decided to freeze some columns or rows in this grid later, for example (because there will be 3 other "grid windows" then). What other windows do we actually get the events from? I am pretty sure ` wxEVT_GRID_CELL_RIGHT_CLICK` should only be generated when clicking on cells, this is the whole point of having this message in addition to generic `wxEVT_RIGHT_CLICK`.